base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java

```diff
@@ -204,8 +204,21 @@
      */
     Iterable<IBond> bonds();
 
+    /**
+     * Get the number of explicit bonds connected to this atom.
+     * @return the total bond count
+     */
     int getBondCount();
 
+    /**
+     * Returns the bond connecting 'this' atom to the provided atom. If the
+     * atoms are not bonded, null is return
+     * @param atom the other atom
+     * @return the bond connecting the atoms
+     * @throws UnsupportedOperationException thrown if the bonds are not known
+     */
+    IBond getBond(IAtom atom);
```

Comment:
So, the CDK design has the idea that an electron contains has one or more atoms, if two or more, it's a bond... there is indeed a lot of code that assumes two-atom bonds, but this method really formalizes that... This is not a arguing against it, and I'm just pondering where we should head with the CDK...

Reply:
Technically it doesn't, but the implemented it is does as I only check the begin an end. The idea of having multiple atoms in an electron container is cute but really not very well thought or implemented. As described before the better way of doing multicenter bonds is as a layer on top of a valid but inaccurate valence bond representation ala IUPAC InChI discussions and Alex Clark's recent blog post.